app/features/notes_generator/core.py

```diff
@@ -0,0 +1,64 @@
+import re
+from typing import Literal, Optional, Dict
+import logging
+from tools import (
+    extract_text_from_file,
+    extract_text_from_url,
+    generate_notes_rag,
+    clean_content
+)
+
+# Dictionary to map input types to corresponding functions
+file_handler_map = {
+    'text': lambda content: clean_content(content),  # Clean content directly
+    'file': extract_text_from_file,
```

Comment:
In the Marvel AI Backend, we are managing files with URLs and specifying the file type. Please, remove this for standardization purposes.